tools/wptrunner/wptrunner/browsers/base.py

```diff
@@ -275,3 +280,117 @@ def __call__(self, line):
         self.logger.process_output(self.pid,
                                    line.decode("utf8", "replace"),
                                    command=" ".join(self.command) if self.command else "")
+
+
+class WebDriverBrowser(Browser):
+    __metaclass__ = ABCMeta
+
+    default_base_path = "/"
+
+    def __init__(self, logger, binary, webdriver_binary, webdriver_args=None,
+                 host="127.0.0.1", port=None, base_path="", env=None, **kwargs):
+        super().__init__(logger)
+
+        if webdriver_binary is None:
+            raise ValueError("WebDriver server binary must be given "
+                             "to --webdriver-binary argument")
+
+        self.logger = logger
+        self.binary = binary
+        self.webdriver_binary = webdriver_binary
+
+        self.host = host
+        self._port = port
+
+        if base_path == "":
+            self.base_path = self.default_base_path
+        else:
+            self.base_path = base_path
+        self.env = os.environ.copy() if env is None else env
+        self._args = webdriver_args if webdriver_args is not None else []
```

Comment:
Since we're juggling two binaries here, maybe:

```suggestion
        self.webdriver_args = webdriver_args if webdriver_args is not None else []
```